drivers/src/sha1.rs

```diff
@@ -244,7 +237,7 @@ impl<'a> Sha1DigestOp<'a> {
     }
 
     /// Finalize the digest operations
-    pub fn finalize(&mut self) -> CaliptraResult<()> {
+    pub fn finalize(&mut self, digest: &mut Array4x5) -> CaliptraResult<()> {
```

Comment:
Can you also have this take `self` or `mut self` instead of `&mut self`? We shouldn't allow people to call finalize multiple times.

Reply:
Done.